server/queues/processors/WebsocketsProcessor.ts

```diff
@@ -278,57 +282,63 @@ export default class WebsocketsProcessor {
       }
 
       case "collections.add_group": {
-        const groupUsers = await GroupUser.findAll({
-          where: { groupId: event.modelId },
-        });
-
-        // the users being added are not yet in the websocket channel for the collection
-        // so they need to be notified separately
-        for (const groupUser of groupUsers) {
-          socketio.to(`user-${groupUser.userId}`).emit("collections.add_user", {
-            event: event.name,
-            userId: groupUser.userId,
-            collectionId: event.collectionId,
-          });
-          // tell any user clients to connect to the websocket channel for the collection
-          socketio.to(`user-${groupUser.userId}`).emit("join", {
-            event: event.name,
-            collectionId: event.collectionId,
-          });
+        const membership = await GroupMembership.findByPk(
+          event.data.membershipId
+        );
+        if (!membership) {
+          return;
         }
 
+        socketio
+          .to(`group-${membership.groupId}`)
+          .emit(event.name, presentGroupMembership(membership));
+
+        socketio
+          .to(`collection-${membership.collectionId}`)
+          .emit(event.name, presentGroupMembership(membership));
+
+        socketio.to(`group-${membership.groupId}`).emit("join", {
+          event: event.name,
+          collectionId: event.collectionId,
+        });
+
         return;
       }
 
       case "collections.remove_group": {
-        const [groupUsers, membershipUserIds] = await Promise.all([
-          GroupUser.findAll({
-            where: { groupId: event.modelId },
-          }),
-          Collection.membershipUserIds(event.collectionId),
-        ]);
+        const groupUsers = await GroupUser.findAll({
+          where: { groupId: event.modelId },
+        });
+
+        const membership = {
+          groupId: event.modelId,
+          collectionId: event.collectionId,
+          id: event.data.membershipId,
+        };
+
+        // let everyone with access to the collection know a group was removed
+        socketio
+          .to(`collection-${event.collectionId}`)
+          .emit("collections.remove_group", membership);
+
+        // let everyone in the group know they were removed
+        socketio
+          .to(`group-${event.modelId}`)
+          .emit("collections.remove_group", membership);
 
         for (const groupUser of groupUsers) {
-          if (membershipUserIds.includes(groupUser.userId)) {
-            // the user still has access through some means...
-            // treat this like an add, so that the client re-syncs policies
-            socketio
-              .to(`user-${groupUser.userId}`)
-              .emit("collections.add_user", {
```

Comment:
Cleaned up this logic a lot, we won't emit `collections.add_user` events when a group is removed from the collection anymore.